app/models/predicate.rb

```diff
@@ -17,4 +17,108 @@
 class Predicate < ApplicationRecord
   validates :uri, presence: true, uniqueness: true
   validates :pref_label, presence: true
+  after_create :assign_color, unless: proc { color.present? }
```

Comment:
A `before_create` callback will be better, b/c it allows assigning a color in the same SQL statements, whereas `after_create` results in two trips to the DB. Also, the proc isn't necessary here:

`before_create :assign_color, unless: :color?`